src/pf-donut-chart/pf-donut-chart.component.js

```diff
@@ -0,0 +1,359 @@
+ * @prop {string} width width of chart
+ * @prop {string} height height of chart
+ * @prop {JSON} legend c3 legend for chart (method takes object as argument)
+ * @prop {string} target-selector target element selector
+ * @prop {string} title title for chart
+ *
+ * @method tooltip takes an object as argument with its properties show and content
+ * @method onclick takes a function as argument -- for c3 data.onclick
+ * @method onmouseover takes a function as argument -- for c3 data.onmouseover
+ * @method onmouseout takes a function as argument -- for c3 data.onmouseout
+ * @method load takes an object as argument -- for c3 api load
+ * @method unload takes an object as argument -- for c3 api unload
+ */
+
+export class PfDonutChart extends HTMLElement {
+
+  /**
+   * Reinitializes with attribute values and resets content
+   */
+  init() {
+    this._additionalData = this.getAttribute('data') ? pfChartUtil.getJSONAttribute(this, 'data') : {};
+    this._width = parseInt(this.getAttribute('width')) ? this.getAttribute('width') : null;
+    this._height = parseInt(this.getAttribute('height')) ? this.getAttribute('height') : 171;
+    this._legend = this.getAttribute('legend') ? pfChartUtil.getJSONAttribute(this, 'legend') : { show: false };
+    this._targetSelector = this.getAttribute('target-selector');
+    this._title = this.getAttribute('title') ? this.getAttribute('title') : '';
+    this._colors = this.getAttribute('colors') ? pfChartUtil.getJSONAttribute(this, 'colors') : {};
+    this._getData();
+    this._prepareData();
+  }
+
+  /**
+   * Called when an instance was inserted into the document
+   */
+  connectedCallback() {
+    this.init();
+  }
+
+  /*
+   * Only attributes listed in the observedAttributes property will receive this callback
+   */
+  static get observedAttributes() {
+    return ['width', 'height', 'legend', 'target-selector', 'title', 'data', 'colors', 'columns', 'rows', 'json', 'url'];
+  }
+
+  /**
+ * Called when element's attribute value has changed
+ *
+ * @param {string} attrName The attribute name that has changed
+ * @param {string} oldValue The old attribute value
+ * @param {string} newValue The new attribute value
+ */
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    this.init();
+  }
+
+  /*
+   * An instance of the element is created or upgraded
+   */
+  constructor() {
+    super();
+  }
+
+  /**
+   * Get chart width
+   */
+  get width() {
+    return this._width;
+  }
+
+  /**
+   * Set chart width
+   */
+  set width(value) {
+    if (this._width !== value) {
+      this._width = value;
+      this.setAttribute('width', value);
+    }
+  }
+
+  /**
+   *  Get chart height
+   */
+  get height() {
+    return this._height;
+  }
+
+  /**
+   * Set chart height
+   */
+  set height(value) {
+    if (this._height !== value) {
+      this._height = value;
+      this.setAttribute('height', value);
+    }
+  }
+
+  /**
+   * Get charts legend
+   */
```

Comment:
Probably haven't called this out before, but placing the "returns" in here rounds your annotations out nicely
```js
* Get charts legend
* @returns {string}
```